src/mixins/displayed-component-mixin.ts

```diff
@@ -41,7 +42,8 @@ export class DisplayedComponentMixin extends Vue {
       await this.cancelAnalyzeName('Tabs')
       this.redirectToStart()
     } else if (['ExistingRequestEdit'].indexOf(componentName) > -1) {
-      // We're editing
+      // Cancel edit mode
+      await this.setEditMode(false)
```

Comment:
This is called if the user clicks Return or Exit during edit mode (or anytime probably, in which case this does nothing).